### PR TITLE
Updated secrets in dev readme

### PR DESCRIPTION
**Pull request by CarolynMabbott**
# Changes

Updated secrets endpoints in dev README including examples 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -294,18 +294,18 @@ Get the logs for a PipelineRun by name
...
 ```
 
-__Credentials__
+__Secrets__
 ```
-GET /v1/namespaces/<namespace>/credentials
-Get all credentials by name in the given namespace
-Returns HTTP code 500 if an error occurred getting the credentials
-Returns HTTP code 200 and the given credential as a Kubernetes secret in the given namespace with a blanked out password if found, otherwise an empty list is returned
+GET /v1/namespaces/<namespace>/secrets
```
Comment (AlanGreene) [inline DEVELOPMENT.md]: Most (all?) of these were replaced with calls to the kube API instead in https://github.com/tektoncd/dashboard/issues/320.

e.g.
- `GET /proxy/api/v1/secrets/?fieldSelector=type%3Dkubernetes.io%2Fbasic-auth`
- `GET /proxy/api/v1/namespaces/<namespace>/secrets/?fieldSelector=type%3Dkubernetes.io%2Fbasic-auth`

 If the back end code supporting the legacy APIs is still present it should be removed along with the docs.

Comment (CarolynMabbott) [reply DEVELOPMENT.md]: The back end code was taken out, these are what the Kube API uses to post / get things to and from

Comment (AlanGreene) [reply DEVELOPMENT.md]: We generally don't document the kube APIs on our side as they're already well documented publicly and we simply proxy through to them using the `/proxy` endpoint we provide.

See previous removal of back end APIs https://github.com/tektoncd/dashboard/commit/ee0bbc2ee9366350d75f20230144bcb20adcd822#diff-f785e9e46b9afd2567df7a658338db17

Comment (CarolynMabbott) [reply DEVELOPMENT.md]: Have done this in https://github.com/tektoncd/dashboard/pull/752

**Pull request closed without merging**